src/Core/Models/Request/SetKeyConnectorKeyRequest.cs

```diff
@@ -9,15 +9,18 @@ public class SetKeyConnectorKeyRequest
         public KeysRequest Keys { get; set; }
         public KdfType Kdf { get; set; }
         public int? KdfIterations { get; set; }
+        public int? KdfMemory { get; set; }
+        public int? KdfParallelism { get; set; }
         public string OrgIdentifier { get; set; }
 
-        public SetKeyConnectorKeyRequest(string key, KeysRequest keys,
-            KdfType kdf, int? kdfIterations, string orgIdentifier)
+        public SetKeyConnectorKeyRequest(string key, KeysRequest keys, KdfConfig kdfConfig, string orgIdentifier)
         {
             this.Key = key;
             this.Keys = keys;
-            this.Kdf = kdf;
-            this.KdfIterations = kdfIterations;
+            this.Kdf = kdfConfig.Type.GetValueOrDefault(KdfType.PBKDF2_SHA256);
```

Comment:
Previously the default value wasn't being set, is it necessary now?

Reply:
This is due to the same reason as above:

> This is due to type and iterations being optional in the kdfconfig struct. Changing them to non-optional in the struct breaks some other things (f.e setting default iteration count in the CryptoService in the MakeKeyAsync function.) and require changes there